datafusion/sql/src/statement.rs

```diff
@@ -385,25 +385,32 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let DescribeTable { table_name } = statement;
 
         let where_clause = object_name_to_qualifier(&table_name);
+        let table_str_name = object_name_to_table_str_name(&table_name);
         let table_ref = object_name_to_table_reference(table_name)?;
 
-        // check if table_name exists
-        let _ = self
+        let table_source = self
             .schema_provider
             .get_table_provider((&table_ref).into())?;
 
-        if self.has_table("information_schema", "tables") {
-            let sql = format!(
-                "SELECT column_name, data_type, is_nullable \
-                                FROM information_schema.columns WHERE {where_clause};"
-            );
-            let mut rewrite = DFParser::parse_sql(&sql)?;
-            self.statement_to_plan(rewrite.pop_front().unwrap())
+        if !table_source.is_temporary_file() {
```

Comment:
I wonder if we can figure out a way to avoid special casing here. Perhaps we can always create a DescribeTable plan?